lib/src/network_request.dart

```diff
@@ -146,6 +146,29 @@ class DeleteRequest extends NetworkRequest {
         );
 }
 
+/// {@template raw_request}
+/// A [NetworkRequest] that allows for passing a [NetworkRequestType] (useful
+/// for when this can't be known until runtime). It also defaults to a raw
+/// (null) body.
+/// {@endtemplate}
+class RawRequest extends NetworkRequest {
+  /// {@macro raw_request}
+  const RawRequest(
+    String path, {
+    required super.type,
+    super.data = const NetworkRequestBody.raw(null),
```

Comment:
Is it particularly useful to have a default value here?

Reply:
I think maybe equally useful as not having it? Hard to say whether majority of cases would rely on the default or provide their own, but requiring it would mean 100% of use cases need to provide it, so I leaned towards the conservative. Chances are if you forget to provide your data it's a pretty obvious flub during development, but needing to provide `NetworkRequestBody.raw(null)` to every request you don't want data for feels burdensome.